evml/metrics.py

```diff
@@ -0,0 +1,624 @@
+        no_skill = [0.5 * x1 + 0.5 * ave_true for x1 in histogram["bin"]]
+        axs[k].plot(histogram["bin"], no_skill, "b-")
+        axs[k].plot(histogram["bin"], ave_true_range, ls=":", color="lightgrey")
+        axs[k].plot(ave_true_range, histogram["mean"], ls=":", color="lightgrey")
+
+        full_range = np.linspace(min(histogram["bin"]), max(histogram["bin"]), 200)
+        no_skill = [0.5 * x1 + 0.5 * ave_true for x1 in full_range]
+        fill_cond = np.where(full_range < ave_true)[0]
+        axs[k].fill_between(
+            np.array(full_range)[fill_cond],
+            min(
+                min(np.array(histogram["mean"]) - np.array(histogram["std"])),
+                min(np.array(full_range)),
+            ),
+            np.array(no_skill)[fill_cond],
+            color="lightblue",
+        )
+        fill_cond = np.where(full_range > ave_true)[0]
+        axs[k].fill_between(
+            np.array(full_range)[fill_cond],
+            max(
+                max(np.array(histogram["mean"]) + np.array(histogram["std"])),
+                max(np.array(full_range)),
+            ),
+            np.array(no_skill)[fill_cond],
+            color="lightblue",
+        )
+
+        axs[k].set_title(f"{legend_cols[k]}")
+        axs[k].set_ylabel("Conditional mean observation")
+        axs[k].set_xlabel("Prediction")
+
+    plt.tight_layout()
+    if save_location:
+        plt.savefig(
+            os.path.join(save_location, "regression_attributes.pdf"),
+            dpi=300,
+            bbox_inches="tight",
+        )
+
+
+
+def compute_pit(true_dist, pred_dist):
+    """
+    Computes the probability integral transform (PIT) between a true distribution
+    and a predicted distribution.
+
+    Parameters
+    ----------
+    true_dist : array-like
+        A 1D array of values representing the true distribution.
+    pred_dist : array-like
+        A 1D array of values representing the predicted distribution.
+
+    Returns
+    -------
+    pit : array-like
+        A 1D array of PIT values.
+    """
+    # Sort the true and predicted distributions
+    true_sorted = np.sort(true_dist)
+    pred_sorted = np.sort(pred_dist)
+
+    # Compute the PIT values
+    pit = np.zeros_like(pred_sorted)
+    for i, x in enumerate(pred_sorted):
+        j = np.searchsorted(true_sorted, x, side="right")
+        pit[i] = j / len(true_sorted)
+
+    return pit
+
+
+def pit_histogram(
+    df, pred_dist, output_cols, num_bins=50, legend_cols=None, save_location=None
+):
+    width = 5 if len(output_cols) == 1 else 10
+    height = 3.5 if len(output_cols) == 1 else 3.5
+    fig, axs = plt.subplots(1, len(output_cols), figsize=(width, height))
+    if len(output_cols) == 1:
+        axs = [axs]
+
+    if legend_cols is None:
+        legend_cols = output_cols
+
+    true_dist = df[output_cols].values
+
+    for k in range(true_dist.shape[-1]):
+        td = true_dist[:, k]
+        pd = pred_dist[:, k]
```

Comment:
For a given output prediction, shouldn't there be more than one ensemble member? The way the slicing is done here makes it look like there is one prediction per example.